src/Neo/SmartContract/Helper.cs

```diff
@@ -78,6 +78,69 @@ public static void Check(byte[] script, ContractAbi abi)
         /// <remarks>Note: The <see cref="Script"/> passed to this method should be constructed with strict mode.</remarks>
         public static void Check(this Script script, ContractAbi abi)
         {
+            if (abi is null) throw new ArgumentNullException(nameof(abi));
+
+            Dictionary<int, Instruction> instructions = new();
+            for (int ip = 0; ip < script.Length;)
+            {
+                Instruction instruction = script.GetInstruction(ip);
+                instructions.Add(ip, instruction);
+                ip += instruction.Size;
+            }
```

Comment:
But isn't the same code executed inside the `Script` constructor? See https://github.com/neo-project/neo-vm/blob/5b0a39811b34abacab1273f3ee5a9a9f7e52ac7b/src/Neo.VM/Script.cs#L72.

Reply:
See also the https://github.com/neo-project/neo/issues/2848#issuecomment-1465860765.

Reply:
You are right 😅

Reply:
The problem is that the default value for strictMode is false

Reply:
Oh, see the https://github.com/neo-project/neo/pull/2849#discussion_r1133726025, please.